File: Wed_Thời Trang Nam/ThuanBackendNode/models/VideoTracking.js
// models/VideoTracking.js
const mongoose = require('mongoose');

const VideoTrackingSchema = new mongoose.Schema({
    userId: { type: Number, required: true }, // ID from MySQL users table
    videoUrl: { type: String, required: true },
    videoTitle: { type: String, default: '' },
    watchedSeconds: { type: Number, required: true },
    totalDuration: { type: Number, required: true },
    watchPercentage: { type: Number, default: 0 },
    interest: { type: String, default: null }, // JSON or string
    watchedAt: { type: Date, default: Date.now },
    startedAt: { type: Date, default: null },
    endedAt: { type: Date, default: null }
});

module.exports = mongoose.model('VideoTracking', VideoTrackingSchema);

File: Wed_Thời Trang Nam/ThuanBackend/Thuan.API/Models/VideoTracking.cs
// Models/VideoTracking.cs
using System.ComponentModel.DataAnnotations;

namespace Thuan.API.Models
{
    public class VideoTracking
    {
        public int Id { get; set; }

        [Required]
        public int UserId { get; set; }

        [Required]
        public string VideoUrl { get; set; } = string.Empty;

        public string? VideoTitle { get; set; } // Tên video

        [Required]
        public int WatchedSeconds { get; set; } // Đoạn xem được (giây)

        [Required]
        public int TotalDuration { get; set; } // Tổng độ dài video (giây)

        public double WatchPercentage { get; set; } // % xem (0-100)

        public string? Interest { get; set; } // Quan tâm gì (JSON: {"color": "đỏ", "style": "casual"})

        public DateTime WatchedAt { get; set; } = DateTime.UtcNow;

        public DateTime? StartedAt { get; set; } // Lúc nào bắt đầu xem

        public DateTime? EndedAt { get; set; } // Lúc nào kết thúc xem
    }
}
